### ISSUE TITLE
Adresser som mangler

### ISSUE THREAD

**Issue by NKAmapper** [rubund/addrnodeimport#24]
Det mangler mange tusen (100.000 ?) adresser som tydeligvis ikke kommer med i importen. Kan noe gjøres?

Her er noen alternative kilder: https://kartkatalog.geonorge.no/search?text=adresse&Facets%5B0%5D.name=organization&Facets%5B0%5D.value=Kartverket&Facets%5B1%5D.name=type&Facets%5B1%5D.value=dataset

Jeg har f.eks. testet tabelluttrekk for et par fylker fra https://kartkatalog.geonorge.no/metadata/kartverket/matrikkelen-adresse/f7df7a18-b30f-4745-bd64-d0863812350c, og får da et stort antall ekstra adresser, inklusive for kommuner som var nesten tomme for adresser i OSM (har sannsynligvis fått gate/husnummer adresse i løpet av de par siste årene).

Ellers er dette et virkelig flott tiltak. Jeg har stor nytte av adressene ved geokoding mot Nominatim for å gjøre andre importer (legekontorer, rådhus mv.)

**Comment by Gazer75** [comment]
Eg fekk ei osm fil for Sogn og Fjordane av NKA og har manuelt lagt til eller oppdatert i nokre av kommunane som hadde største manglar.
For Bremanger så mangla alle. Solund, Høyanger, Hyllestad, Fjaler, Flora og Eid hadde litt rundt kommunesenteret, men alle mangla mesteparten av adressene utanfor desse. Skulle tippe 40-60% av adressene mangla for desse kommunane.
Om eg skulle tippe så har eg vel lagt til mellom 5 og 10 tusen adresser.

Sogndal har eg lagt til heilt manuelt tidlegare og der var det også sikkert 20-30% som manglar totalt. Akkurat for den kommunen så har eg ikkje teke området rundt Fjærland endå.

Om det står like ille til i andre fylke så manglar det nok fleire titals tusen adresser i OSM.

Om det hadde gått an å ta ein manuell nedlasting av data for å i det minste oppdatere data ein gong så ville det hjelpe fram til REST API er oppe att og oppdatert. Kartverket seier at det skal vere i orden at før jol.

Eit anna problem er jo at der kommunar vart slått saman etter 2016 så har fleire gater endra navn, og dette har nok heller ikkje vorte oppdatert.

**Comment by NKAmapper** [comment]
Det har kommet en ny versjon av et adresse-api fra Kartverket: [Adresse REST-API (test)](https://kartkatalog.geonorge.no/metadata/kartverket/adresse-rest-api-test/44eeffdc-6069-4000-a49b-2d6bfc59ac61)

Var det den gamle versjonen av dette som ble brukt til å hente adressene?

**Comment by ghost** [comment]
Det ryktes om at dette bare er et test-API, og at det kommer et endelig API etter jul. Uansett så fungerer dette nye kjempefint. Jeg bruker det på jobb

**Comment by NKAmapper** [comment]
Det står på [produktsiden](https://ws.geonorge.no/adresser/test/) at api'et bare returnerer de første 10.000 adressene, så dette ser ikke ut til å fungere for kommuner med mer enn 15-20.000 innbyggere.

Et godt alternativ er å hente ned adressene i ferdige filer [her](https://nedlasting.geonorge.no/geonorge/Basisdata/MatrikkelenVegadresse/CSV/).

**Comment by rubund** [comment]
Dette APIer blir brukt nå: http://ws.geonorge.no/AdresseWS/adresse/boundingbox?nordLL=58.9&austLL=6.3&nordUR=59.0&austUR=6.4

Det nye virker jo ubrukelig siden det returnerer maks 10.000 + at jeg ikke finner muligheten til å be om adresser innenfor en boundingbox. Det første var jo helt genialt - så hadde det bare funka ..

@NKAmapper  Blir https://nedlasting.geonorge.no/geonorge/Basisdata/MatrikkelenVegadresse/CSV/ holdt oppdatert.

Det kan være mulig å bruke, men da vil jeg laste ned hele samlinga før jeg etterpå splitter med boundingbokser.

**Comment by NKAmapper** [comment]
Ja, [Vegdresse](https://nedlasting.geonorge.no/geonorge/Basisdata/MatrikkelenVegadresse/CSV/) holdes løpende oppdatert.

Jeg laget et nytt importprogram [her](https://github.com/NKAmapper/addr2osm), så da behøves vel egentlig ingen oppdatering. Det har blitt diskutert på maillisten og er kjørt et par ganger nå for hele landet.

**Comment by rubund** [comment]
@NKAmapper Supert. Da slipper jeg å ha dårlig samvittighet for dette (jeg er svært lite OSM-aktiv for tida). Det er bra at du skar igjennom med bare å putte noder på alle adresser. Det ville gjort det mye enklere for meg også noen år tilbake da jeg begynte med dette. (men da var ikke tiden helt moden for det).
Kos deg med stafettpinnen.
